pylint/reporters/reports_handler_mix_in.py

```diff
@@ -24,12 +24,12 @@ def __init__(self) -> None:
         self._reports: ReportsDict = collections.defaultdict(list)
         self._reports_state: Dict[str, bool] = {}
 
-    def report_order(self) -> List[checkers.BaseChecker]:
+    def report_order(self) -> List["BaseChecker"]:
```

Comment:
The return type should probably be `MutableSequence["BaseChecker"]` instead of `List`. The later is invariant, so technically the list can only contain `BaseChecker` classes. `MutableSequence` is covariant, thus subclasses are also allowed.

Reply:
8b00b6e